src/test/java/org/terracotta/offheapstore/disk/storage/FileBackedStorageEngineTest.java

```diff
@@ -123,6 +130,61 @@ public void testInterruptingReadThreads() throws IOException {
     }
   }
 
+  @Test
+  public void testConcurrentReads() throws IOException {
+    MappedPageSource source = new MappedPageSource(dataFile);
```

Comment:
Not directly related but may be useful, and rules out another potential cause of concurrency issues.